tests/contenttypes_tests/test_fields.py

```diff
@@ -33,3 +36,19 @@ def test_value_to_string(self):
         answer2 = Answer.objects.create(question=question)
         result = json.loads(Question.answer_set.field.value_to_string(question))
         self.assertCountEqual(result, [answer1.pk, answer2.pk])
+
+    def test_can_get_path_to_proxy_model(self):
+        field = GenericRelationProxyModel._meta.get_field('proxy_model')
+        self.assertEqual(
+            field.get_path_info(),
+            [
+                models.query_utils.PathInfo(
+                    from_opts=GenericRelationProxyModel._meta,
+                    to_opts=GenericConcreteModel._meta,
+                    target_fields=(GenericConcreteModel._meta.get_field('id'),),
+                    join_field=field.remote_field,
+                    m2m=True,
+                    direct=False
+                )
+            ]
+        )
```

Comment:
It would be great to have a regression test for the `AttributeError` raised by `filter()`'ing as well.